pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ParserTstUtil.java

```diff
@@ -112,31 +120,104 @@ public ASTCompilationUnit buildDFA(String javaCode) {
     }
 
-    public ASTCompilationUnit parseJava(LanguageVersion languageVersion, String code) {
-        LanguageVersionHandler languageVersionHandler = languageVersion.getLanguageVersionHandler();
+    /** @see #parseJava(LanguageVersionHandler, String)  */
+    public static ASTCompilationUnit parseJava13(Class<?> source) {
+        return parseJava13(getSourceFromClass(source));
+    }
+
+    /** @see #parseJava(LanguageVersionHandler, String)  */
+    public static ASTCompilationUnit parseJava14(Class<?> source) {
+        return parseJava14(getSourceFromClass(source));
+    }
+
+    /** @see #parseJava(LanguageVersionHandler, String)  */
+    public static ASTCompilationUnit parseJava15(Class<?> source) {
+        return parseJava15(getSourceFromClass(source));
+    }
+
+    /** @see #parseJava(LanguageVersionHandler, String)  */
+    public static ASTCompilationUnit parseJava17(Class<?> source) {
+        return parseJava17(getSourceFromClass(source));
+    }
+
+    /** @see #parseJava(LanguageVersionHandler, String)  */
+    public static ASTCompilationUnit parseJava18(Class<?> source) {
+        return parseJava18(getSourceFromClass(source));
+    }
+
+    /** @see #parseJava(LanguageVersionHandler, String) */
+    public static ASTCompilationUnit parseJavaDefaultVersion(String source) {
+        return parseJava(getDefaultLanguageVersionHandler(), source);
+    }
+
+    /** @see #parseJava(LanguageVersionHandler, String) */
+    public static ASTCompilationUnit parseJavaDefaultVersion(Class<?> source) {
+        return parseJavaDefaultVersion(getSourceFromClass(source));
+    }
+
+
+    public static LanguageVersionHandler getLanguageVersionHandler(String version) {
+        return LanguageRegistry.getLanguage(JavaLanguageModule.NAME).getVersion(version).getLanguageVersionHandler();
+    }
+
+    public static LanguageVersionHandler getDefaultLanguageVersionHandler() {
+        return LanguageRegistry.getLanguage(JavaLanguageModule.NAME).getDefaultVersion().getLanguageVersionHandler();
+    }
+
+
+    /**
+     * Parses Java code and executes the symbol table visitor.
+     *
+     * @param languageVersionHandler The version handler for the wanted version
+     * @param code                   The source code
+     *
+     * @return The compilation unit
+     */
+    public static ASTCompilationUnit parseJava(LanguageVersionHandler languageVersionHandler, String code) {
         ASTCompilationUnit rootNode = (ASTCompilationUnit) languageVersionHandler
                 .getParser(languageVersionHandler.getDefaultParserOptions()).parse(null, new StringReader(code));
         languageVersionHandler.getSymbolFacade().start(rootNode);
         return rootNode;
     }
+
+    public static String getSourceFromClass(Class<?> clazz) {
+        String sourceFile = clazz.getName().replace('.', '/') + ".java";
+        InputStream is = ParserTstUtil.class.getClassLoader().getResourceAsStream(sourceFile);
```

Comment:
Also, we have configuration on poms to support it under particular packages (ie: type resolution!). See: https://github.com/pmd/pmd/blob/master/pmd-java/pom.xml#L22-L27

Otherwise, it doesn't work, at least not from Maven.